MedicalDocManagment.WebUI/app/components/childrens-card/diagnosis/select-modal.component.ts

```diff
@@ -0,0 +1,277 @@
+
+        this._blocksLoading = this._classesLoading = this._diagnosesLoading =
+            this._diagnosisLoading = this._nosologiesLoading = false;
+
+        this._ngZone = ngZone;
+
+        this._resultDiagnosisId = this._selectedBlockIdValue = this._selectedClassIdValue =
+            this._selectedDiagnosisIdValue = this._selectedNosologyIdValue = null;
+
+        this._mkhsService = mkhsService;
+    }
+
+    @Input()
+    get diagnosisId(): string {
+        return this._resultDiagnosisId;
+    }
+    set diagnosisId(value: string) {
+        this._resultDiagnosisId = value;
+        this._loadMkhTreeForDiagnosis(this._resultDiagnosisId);
+
+        this.diagnosisIdChange.emit(value);
+    }
+
+    _diagnosisSelected(): void {
+        this.diagnosisId = this._selectedDiagnosisId;
+    }
+
+    @ViewChild('diagnosisSelectModal')
+    set _diagnosisSelectModal(elementRef: ElementRef) {
+        $(elementRef.nativeElement).on('hide.bs.modal', () => {
+            this._ngZone.run(() => {
+                if (this._selectedDiagnosisId != this.diagnosisId) {
+                    this._loadMkhTreeForDiagnosis(this.diagnosisId);
+                }
+            })
+        });
+    }
+
+    _loadClasses(afterLoad: () => void = () => { }): void {
+        this._classesLoading = true;
+        this._classes = [];
+        var subscription = this._mkhsService.getClasses()
+            .subscribe(
+            (classes: ClassesModel) => {
+                this._classes = this._classes.concat(classes);
+            },
+            (error) => { },
+            () => {
+                this._classesLoading = false;
+                afterLoad();
+                subscription.unsubscribe();
+            })
+    }
+
+    _loadBlocksForClass(classId: string, afterLoad: () => void = () => { }): void {
+        this._blocks = [];
+        if (classId) {
+            this._blocksLoading = true;
+            var subscription = this._mkhsService.getBlocksByClassId(classId)
+                .subscribe(
+                (blocks: BlocksModel) => {
+                    this._blocks = this._blocks.concat(blocks);
+                },
+                (error) => { },
+                () => {
+                    this._blocksLoading = false;
+                    afterLoad();
+                    subscription.unsubscribe();
+                })
+        }
+        else {
+            afterLoad();
+        }
+    }
+
+    _loadDiagnosesForNosology(nosologyId: string, afterLoad: () => void = () => { }): void {
+        this._diagnoses = [];
+        if (nosologyId) {
+            this._diagnosesLoading = true;
+            var subscription = this._mkhsService.getDiagnosesByNosologyId(nosologyId)
+                .subscribe(
+                (diagnoses: DiagnosesModel) => {
+                    this._diagnoses = this._diagnoses.concat(diagnoses);
+                },
+                () => { },
```

Comment:
The purpose of this line of code is to give to the subscribe method with empty body.